### PR TITLE
fix(client): preserve SDK retry behavior in NemoClient adapters

### PR DESCRIPTION
**Pull request by maxdubrinsky**
## What

Preserve the SDK's retry behavior in the generic `NemoClient` adapter layer. Split out of the AIRCORE-876 Models typed-client migration as an independent, low-risk client-layer change.

The generic adapter now preserves the SDK retry policy:
- Stainless retry-decision headers are honored.
- `Retry-After` headers are honored, with a fallback when the value is unreasonable.
- The standalone (non-Stainless) policy keeps its narrower retryable status set (`502, 503, 504, 429`).

## Why split

The AIRCORE-876 branch (`aircore-876-migrate-models-to-nemoclient/md`) is a large, vendoring-gated migration. This retry hardening is conceptually independent of the models work and benefits every `NemoClient` consumer, so it is carved off to shrink the review surface of the big PR. It can merge in any order relative to the typed-models-client PR (disjoint files).

## Scope

Client layer only, no consumer behavior changes:
- `client/adapter.py`, `client/client.py`, `client/types.py`
- Tests: `tests/client/test_adapter.py`, `tests/client/test_client_options.py`

## Verification

- `pytest packages/nemo_platform_plugin/tests` -> 1042 passed
- `ruff check` / `ruff format --check` clean

## Summary by CodeRabbit

* **New Features**
  * Expanded retry policy support for retryable HTTP statuses and optional server-error retries.
  * Added configurable handling for retry-decision and `Retry-After` headers, including supported delay formats and safe limits.

* **Bug Fixes**
  * Improved retry timing with exponential backoff and consistent behavior for synchronous and asynchronous requests.

* **Tests**
  * Expanded coverage for retry defaults, header-driven behavior, delay parsing, fallback handling, and edge cases.

### CODE REVIEW

File: packages/nemo_platform_plugin/src/nemo_platform_plugin/client/client.py [hunk 1]
```diff
@@ -117,6 +119,28 @@ def _get_paginated_types(
...
+    except (TypeError, ValueError):
+        retry_after = response.headers.get("retry-after")
+        try:
+            delay = float(retry_after)
+        except (TypeError, ValueError):
+            # Retry-After may instead be an RFC 5322 / HTTP-date, e.g.
+            # "Fri, 31 Dec 2027 23:59:59 GMT"; convert that to a delta.
+            try:
+                retry_date = email.utils.parsedate_to_datetime(retry_after)
+            except (TypeError, ValueError):
```
Comment (marcusds) [inline packages/nemo_platform_plugin/src/nemo_platform_plugin/client/client.py]: Do we want to catch OverflowError incase the retry_after is a massive date or something silly?

**Review by marcusds — APPROVED** [review]
